tests/test_api.py

```diff
@@ -0,0 +1,195 @@
+        "filters": [],
+        "skip": 0,
+    }
+
+    response = await joint_fixture.rest_client.post(
+        url="/rpc/search", json=search_parameters
+    )
+    assert response.status_code == 500
+
+
+@pytest.mark.asyncio
+async def test_search(joint_fixture: JointFixture):  # noqa: F811
+    """Basic query to pull back all documents for class name"""
+    search_parameters: JsonObject = {
+        "class_name": "DatasetEmbedded",
+        "query": "",
+        "filters": [],
+        "skip": 0,
+    }
+
+    results = await call_search(search_parameters, joint_fixture)
+    compare(results=results, count=3, hit_length=3)
+
+
+@pytest.mark.asyncio
+async def test_search_with_limit(joint_fixture: JointFixture):  # noqa: F811
+    """Make sure we get a count of 3 but only 1 hit"""
+    search_parameters: JsonObject = {
+        "class_name": "DatasetEmbedded",
+        "query": "",
+        "filters": [],
+        "skip": 0,
+        "limit": 1,
+    }
+
+    results = await call_search(search_parameters, joint_fixture)
+    hit = {
+        "id_": "1HotelAlpha-id",
+        "content": {
+            "category": "hotel",
+            "field1": "Miami",
+            "has_object": {"id_": "HotelAlphaObject", "type": "piano"},
+            "has_rooms": [
+                {"id_": "HotelAlphaLarge", "type": "large room"},
+                {"id_": "HotelAlphaPoolside", "type": "poolside room"},
+            ],
+            "type": "resort",
+        },
+    }
+    hits = [models.Resource(**hit)]
+    compare(results=results, count=3, hit_length=1, hits=hits)
+
+
+@pytest.mark.asyncio
+async def test_search_keywords(joint_fixture: JointFixture):  # noqa: F811
+    """Make sure the query string is passed through intact"""
+    search_parameters: JsonObject = {
+        "class_name": "DatasetEmbedded",
+        "query": "hotel",
+        "filters": [],
+        "skip": 0,
+    }
+
+    results = await call_search(search_parameters, joint_fixture)
+    compare(results=results, count=2, hit_length=2)
+
+
+@pytest.mark.asyncio
+async def test_search_filters(joint_fixture: JointFixture):  # noqa: F811
+    """Make sure filters work"""
+    search_parameters: JsonObject = {
+        "class_name": "DatasetEmbedded",
+        "query": "",
+        "filters": [{"key": "has_object.type", "value": "piano"}],
+        "skip": 0,
+    }
+
+    results = await call_search(search_parameters, joint_fixture)
+    compare(results=results, count=1, hit_length=1)
+
+
+@pytest.mark.asyncio
+async def test_search_invalid_class(joint_fixture: JointFixture):  # noqa: F811
+    """Verify that searching with a bad class name results in a 404"""
+    search_parameters: JsonObject = {
+        "class_name": "InvalidClassName",
+        "query": "",
+        "filters": [],
+        "skip": 0,
+        "limit": 1,
+    }
+
+    response = await joint_fixture.rest_client.post(
+        url="/rpc/search", json=search_parameters
+    )
+    assert response.status_code == 404
```

Comment:
```suggestion
    assert response.status_code == 422
```

Reply:
Fixed in [072b380](https://github.com/ghga-de/mass/pull/7/commits/072b3800ef9b2ac017a691a68f1df8e15ea8d287)